WordPress/src/main/java/org/wordpress/android/ui/stats/StatsGeoviewsFragment.java

```diff
@@ -20,6 +31,121 @@
+
+        mTopPagerContainer.setVisibility(View.GONE);
+    }
+
+    @SuppressLint("NewApi")
+    private void showMap(final List<GeoviewModel> countries) {
+        if (!isAdded()) {
+            return;
+        }
+
+        // removeOnGlobalLayoutListener is available on >= JELLY_BEAN. The map is hidden on older devices.
+        if (Build.VERSION.SDK_INT < Build.VERSION_CODES.JELLY_BEAN) {
+            mTopPagerContainer.setVisibility(View.GONE);
+            return;
+        }
+
+        // setting up different margins for the map. We're basically remove left margins since the
+        // chart service produce a map that's slightly shifted on the right. See the Web version.
+        int dp4 = DisplayUtils.dpToPx(mTopPagerContainer.getContext(), 4);
+        LinearLayout.LayoutParams layoutParams = new LinearLayout.LayoutParams(
+                LinearLayout.LayoutParams.MATCH_PARENT, LinearLayout.LayoutParams.WRAP_CONTENT);
+        layoutParams.setMargins(0, 0, dp4, 0);
+        mTopPagerContainer.setLayoutParams(layoutParams);
+
+        mTopPagerContainer.removeAllViews();
+
+        // must wait for mTopPagerContainer to be fully laid out (ie: measured). Then we can read the width and
+        // calculate the right height for the map div
+        mTopPagerContainer.getViewTreeObserver().addOnGlobalLayoutListener(new ViewTreeObserver.OnGlobalLayoutListener() {
+            @Override
+            public void onGlobalLayout() {
+                mTopPagerContainer.getViewTreeObserver().removeOnGlobalLayoutListener(this);
+
+                StringBuilder dataToLoad = new StringBuilder();
+
+                for (int i = 0; i < countries.size(); i++) {
+                    final GeoviewModel currentCountry = countries.get(i);
+                    dataToLoad.append("['").append(currentCountry.getCountryFullName()).append("',")
+                            .append(currentCountry.getViews()).append("],");
+                }
+
+                // This is the label that is shown when the user taps on a region
+                String label = getResources().getString(getTotalsLabelResId());
+
+                // See: https://developers.google.com/chart/interactive/docs/gallery/geochart
+                StringBuilder htmlPage = new StringBuilder().append("<html>")
+                        .append("<head>")
+                        .append("<script type=\"text/javascript\" src=\"https://www.google.com/jsapi\"></script>")
+                        .append("<script type=\"text/javascript\">")
+                            .append("google.load(\"visualization\", \"1\", {packages:[\"geochart\"]});")
+                            .append("google.setOnLoadCallback(drawRegionsMap);")
+                            .append("function drawRegionsMap() {")
+                                .append("var data = google.visualization.arrayToDataTable(" +
+                                        "[")
+                                .append("['Country', '" + label + "'],"
```

Comment:
Minor nit: this does string concatenation with `+` when performance-wise it would be better to use `append()`.